File: rename.py
import os
import re

search_dir = r"d:\AI-Projects\studybuddy-ai\studybuddy-ai"

def replace_func(match):
    word = match.group(0)
    if word == "Learnixio": return "ZenithStudy"
    if word == "learnixio": return "zenithstudy"
    if word == "LEARNIXIO": return "ZENITHSTUDY"
    if word[0].islower(): return "zenithstudy"
    return "ZenithStudy"

ignored_dirs = {".git", "node_modules", ".next"}
ignored_files = {"package-lock.json"}

for root, dirs, files in os.walk(search_dir):
    dirs[:] = [d for d in dirs if d not in ignored_dirs]
    for file in files:
        if file in ignored_files:
            continue
            
        if file.endswith(('.ts', '.tsx', '.js', '.jsx', '.json', '.md', '.sql', '.html', '.css', '.txt')):
            filepath = os.path.join(root, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                new_content = re.sub(r"(?i)learnixio", replace_func, content)
                
                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated {filepath}")
            except Exception as e:
                print(f"Error processing {filepath}: {e}")

print("Done")

File: rename.js
const fs = require('fs');
const path = require('path');

const searchDir = __dirname;
const ignoredDirs = new Set(['.git', 'node_modules', '.next']);
const ignoredFiles = new Set(['package-lock.json', 'rename.js', 'rename.py']);

function replaceFunc(match) {
    if (match === "Learnixio") return "ZenithStudy";
    if (match === "learnixio") return "zenithstudy";
    if (match === "LEARNIXIO") return "ZENITHSTUDY";
    if (match[0] === match[0].toLowerCase()) return "zenithstudy";
    return "ZenithStudy";
}

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        if (ignoredDirs.has(file)) return;
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(filePath));
        } else {
            if (!ignoredFiles.has(file) && file.match(/\.(ts|tsx|js|jsx|json|md|sql|html|css|txt)$/)) {
                results.push(filePath);
            }
        }
    });
    return results;
}

const files = walk(searchDir);

files.forEach(filePath => {
    try {
        const content = fs.readFileSync(filePath, 'utf-8');
        const regex = /learnixio/gi;
        if (content.match(regex)) {
            const newContent = content.replace(regex, replaceFunc);
            fs.writeFileSync(filePath, newContent, 'utf-8');
            console.log(`Updated ${filePath}`);
        }
    } catch (e) {
        console.error(`Error processing ${filePath}: ${e}`);
    }
});

console.log("Done");
